CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+        if(HB_${PKG_NAME}_ROOT_DIRECTORIES AND NOT ${PKG_NAME}_ROOT_DIR)
+            SET(${PKG_NAME}_ROOT_DIR ${HB_${PKG_NAME}_ROOT_DIR} PARENT_SCOPE)
+        endif(HB_${PKG_NAME}_ROOT_DIRECTORIES AND NOT ${PKG_NAME}_ROOT_DIR)
+    endif((NOT ${PKG_NAME}_INCLUDE_DIR) OR (NOT ${PKG_NAME}_LIBRARIES))
+endfunction(GetRootDirForHomebrew PKG_NAME)
+
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+    GetRootDirForHomebrew(OPENSSL)
+    GetRootDirForHomebrew(CHECK)
+    GetRootDirForHomebrew(JANSSON)
+endif(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+
+find_package(OpenSSL 1.0.1 REQUIRED)
+find_package(Jansson 2.3 REQUIRED)
+find_package(Check 0.9.4)
+
+find_library(RT NAMES rt)
+if (${RT} STREQUAL RT-NOTFOUND)
+	unset(${RT})
+endif(${RT} STREQUAL RT-NOTFOUND)
+
+find_library(M NAMES m)
+if (${M} STREQUAL M-NOTFOUND)
+	unset(${M})
+endif(${M} STREQUAL M-NOTFOUND)
+
+check_type_size(ssize_t SSIZE_T)
+
+# Enable test if possible
+if (NOT CHECK_FOUND OR ENABLE_Tests==OFF)
+	SET(ENABLE_Tests OFF)
+endif(NOT CHECK_FOUND OR ENABLE_Tests==OFF)
+message(STATUS "Tests Enabled............${ENABLE_Tests}")
+
+CHECK_SYMBOL_EXISTS("random" stdlib.h HAVE_RANDOM)
+
+# Set dependency include directories
+include_directories(
+${OPENSSL_INCLUDE_DIR}
+${JANSSON_INCLUDE_DIRS}
+)
+
+#Generate config file
+configure_file (
+  "${PROJECT_SOURCE_DIR}/include/cjose/config.h.in"
+  "${PROJECT_BINARY_DIR}/include/cjose/cjose_config.h"
+)
+
+# Setup compilation flags for different compilers
+if (MSVC)
+   add_compile_options(/W2  /nologo /WX-)
+   add_definitions( "/DVERSION=\"${PACKAGE_VERSION}\"")
+   add_definitions( "/D_CRT_SECURE_NO_WARNINGS")
+   set(DEFAULT_COMPILER_OPTIONS /Zi /DEBUG $<$<CONFIG:Debug>:/Od> $<$<CONFIG:Release>:/O2> /MP )
+   set(LIB_COMPILER_OPTIONS ${DEFAULT_COMPILER_OPTIONS} /WX)
+   if (STATIC_CRT)
+      SET(LIB_COMPILER_OPTIONS ${LIB_COMPILER_OPTIONS} $<$<CONFIG:Debug>:/MTd> $<$<CONFIG:Release>:/MT>)
+   else()
+      SET(LIB_COMPILER_OPTIONS ${LIB_COMPILER_OPTIONS} $<$<CONFIG:Debug>:/MDd> $<$<CONFIG:Release>:/MD>)
+   endif()
+elseif(CMAKE_COMPILER_IS_GNUCC)
+	add_definitions( "-DVERSION=\"${PACKAGE_VERSION}\"")
+    set(DEFAULT_COMPILER_OPTIONS -std=gnu99 -g $<$<CONFIG:Release>:-O2>)
+	set(LIB_COMPILER_OPTIONS ${DEFAULT_COMPILER_OPTIONS} --pedantic -Wall -Werror)
+else(CMAKE_C_COMPILER_ID MATCHES "AppleClang")
+	add_definitions( "-DVERSION=\"${PACKAGE_VERSION}\"")
+    set(DEFAULT_COMPILER_OPTIONS -std=gnu99 -g $<$<CONFIG:Release>:-O2>)
+	set(LIB_COMPILER_OPTIONS ${DEFAULT_COMPILER_OPTIONS} --pedantic -Wall -Werror)
+endif(MSVC)
+
+
+file(GLOB CJOSE_SRC ${PROJECT_SOURCE_DIR}/src/*.c)
+file(GLOB CJOSE_HDR_PUBLIC ${PROJECT_SOURCE_DIR}/include/cjose/*.h)
+file(GLOB CJOSE_HDR_PRIVATE ${PROJECT_SOURCE_DIR}/src/include/*.h)
+
+source_group("Library Sources" FILES ${CJOSE_SRC})
+source_group("Library Private Headers" FILES ${CJOSE_HDR_PRIVATE})
+source_group("Library Public Headers" FILES ${CJOSE_HDR_PUBLIC})

```

Comment:
This is for solution build systems like VisualStudio or xcode. This will create named filters instead of default ones.